server/channels/store/sqlstore/store.go

```diff
@@ -230,13 +241,13 @@ func New(settings model.SqlSettings, metrics einterfaces.MetricsInterface) *SqlS
 
 // SetupConnection sets up the connection to the database and pings it to make sure it's alive.
 // It also applies any database configuration settings that are required.
-func SetupConnection(connType string, dataSource string, settings *model.SqlSettings) *dbsql.DB {
+func SetupConnection(connType string, dataSource string, settings *model.SqlSettings, attempts int) (*dbsql.DB, error) {
 	db, err := dbsql.Open(*settings.DriverName, dataSource)
 	if err != nil {
-		mlog.Fatal("Failed to open SQL connection to err.", mlog.Err(err))
+		return nil, errors.Wrap(err, "failed to open SQL connection")
 	}
 
-	for i := 0; i < DBPingAttempts; i++ {
+	for i := 0; i < attempts; i++ {
```

Comment:
Is there any point of the attempts here as we returning `dbsql.Open` regardless of what atempt number is.

Reply:
From the documentation:

> // Open may just validate its arguments without creating a connection
// to the database. To verify that the data source name is valid, call
// Ping.

So basically, `sql.Open` might not actually do anything worth retrying. `Ping` is the main thing.

Reply:
TIL, thanks for clarification.